src/Discord/Parts/Channel/Channel.php

```diff
@@ -971,4 +971,17 @@ public function getRepositoryAttributes(): array
             'channel_id' => $this->id,
         ];
     }
+    
+    /**
+     * Returns a formatted mention for text channel or name of the channel.
+     *
+     * @return string A formatted mention for text channel or name of the channel.
+     */
+    public function __toString(): string
+    {
+        if ($this->type == self::TYPE_TEXT) {
```

Comment:
Why only for text channels? Should work for all guild channels (text, voice, announcements etc).

Reply:
I wasn't sure about other types, Category was just giving plain text like `#Text Channel`, but yes voice works. I guess it should be fine for all types, I just tested that discord will automatically turn invalid channel id into "#deleted-channel"